README.md

```diff
@@ -1,222 +1,515 @@
+    };
+
+    // The function is bound to the given context.
+    getMessage = getMessage.bind(this);
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
 ```
 
-## Unused Variables
+```javascript
+class MyClass {
+  constructor() {
+    this.message = 'Hello World!';
+  }
 
-Unused variables should be removed from the code when they are no longer needed. This will be [enforced by jshint](http://www.jshint.com/docs/options/#unused).
+  fetchFoo: function(){
+    // The message is assigned to a local variable
+    const message = this.message
 
-Because some libraries expect function arguments with specific names and signatures, function parameters that are not used will not be ignored.
+    const getMessage = function(){
+      return message;
+    };
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
+```
+
+### Curly Braces
+
+Curly braces are required around blocks in loops and conditionals.
 
 **Don't do this:**
+
 ```javascript
-var x;  //x is never used
+if (foo)
+  console.log('bar');
 ```
-**But this is okay:**
+
+**Do this instead:**
+
 ```javascript
-define(function(require, exports, module){
-  //module is never used
-});
+if (foo) {
+  console.log('bar');
+}
 ```
 
-## Coerced Types and Equality
+### Strict Mode
+
+* All modules should be have strict mode enabled.
+* ES6 modules have strict mode enabled by default.
+* Non-ES6 modules should have `'use strict';` placed at the top of the file.
+
+### Coerced Types and Equality
+
+Strict comparison (`===`) should be used instead of abstract comparison (`==`).
 
-Prefer `===` over `==`. This will be [enforced by JShint](http://www.jshint.com/docs/options/#eqeqeq).
+### Trailing Whitespace
 
-## Undefined Variables
+Lines shouldn't have whitespace at the end.
 
-All variables must be explicitly defined. This option will be [enforced by JSHint](http://www.jshint.com/docs/options/#undef).
+### 80 Column Limit
 
-This is a simple way to catch mistyped variable names and copy/paste errors, but it also helps to prevent over-dependence on the global namespace, and forces modules to be more explicit about their dependencies.
+Lines should not exceed 80 characters.
 
-One specific example of why this can be important is in Backbone views, which use a scoped `$` function to find elements inside the view. If you accidentally use the global `$('.selector')` instead of `this.$('.selector')`, you're searching the full DOM instead of just your view.
+In specific cases, long lines can be ignored by the linter.
 
-## Using Variables Before Definition
+**Don't do this:**
 
-Variables must be defined earlier in the code than they are used. (Hoisting makes it possible to accidentally use a variable before it's defined.) This will be [enforced by JSHint](http://www.jshint.com/docs/options/#latedef).
+```javascript
+let longTestPermalinkUrl = 'https://example.com/v1/en/2016/03/14/assets/images/1234.jpg';
+```
 
-## Trailing Whitespace
+**This is allowed:**
```

Comment:
Yeah, it doesn't make much sense in this spot. The "In specific cases" is enough.